extensions/wikia/ArticleVideo/scripts/article-featured-video.js

```diff
@@ -84,22 +89,42 @@ require([
 				};
 
 			if (ooyalaAdSetProvider.canShowAds()) {
-				options.adSet = ooyalaAdSetProvider.get(1, correlator, {
-					contentSourceId: videoData.dfpContentSourceId,
-					videoId: videoId
-				});
 				options.replayAds = ooyalaAdSetProvider.adsCanBePlayedOnNextVideoViews();
+
+				if (a9 && context.bidders && context.bidders.a9Video) {
+					a9.waitForResponse()
+						.then(function () { return a9.getSlotParams('FEATURED'); })
+						.catch(function () { return {}; })
+						.then(function (additionalSlotParams) { // finally
+							options.adSet = setupFirstAdSet(additionalSlotParams);
+							initPlayerWithTracking(options, onCreate);
+						});
+				} else {
+					options.adSet = setupFirstAdSet();
+					initPlayerWithTracking(options, onCreate);
```

Comment:
those 2 lines are almost the same as those from l99, what do you think about extracting it to some function?

Reply:
I considered that, but I think it's cleaner to keep it like this, in other case state of `option` object will be modified from external function